frontend/src/app/core/services/toast-notification.service.ts

```diff
@@ -0,0 +1,65 @@
+import {Injectable} from '@angular/core';
+import {Observable, Subject} from "rxjs";
+import {Message} from "primeng/api";
+
+@Injectable({
+    providedIn: 'root'
+})
+export class ToastNotificationService {
+    private subject = new Subject<Message>();
```

Comment:
Why did you use Subject? You can invoke MessageService directly. In this case, you don't need "toast-notification" component

Reply:
MessageService works only in component and create notifications in "p-toast" component in html template. MessageService can't create own template like a AngularMaterials SnackBar

Reply:
I can remove all this wrappers and developers will always inject MessageService with PrimeNgConfig. But i don't know, maybe notifications will duplicate (don't really know).

Reply:
I googled some solutions for my case. But nothing. All just use directly MessageService.